### PR TITLE
Send a minimum of 1 coin hour when using auto share hours

### LINKED ISSUE #1604: Send a minimum of 1 coin hour when using auto share hours
In `POST /wallet/transaction`, send a minimum of 1 coin hour to all outputs when using auto share hours, if possible.  If there are not enough hours to send a minimum of 1 coin hour to all outputs, it is not an error.

In general the client should avoid sending 0 hours to destinations.  However, if they specify say `0.01` share factor, its possible they will send 0 hours when they could have easily sent at least 1 coin hour.

### PR DESCRIPTION
**Pull request by iketheadore**
Fixes #1604 

Changes:
- Update the POST `/wallet/transaction` endpoint to send a minimum of 1 coin hour to all outputs when possible. 

Does this change need to mentioned in CHANGELOG.md?
Yes

### CODE REVIEW

File: src/transaction/create.go [hunk 1]
```diff
@@ -144,6 +144,19 @@ func create(p Params, auxs coin.AddressUxOuts, headTime uint64, callCount int) (
 				toCoins[i] = to.Coins
 			}
 
+			// If sharefactor > 0 and sharehours < len(p.To), the sharehours cannot ensure each destination
+			// address has at least 1 coin hour. In this case, if we have enough hours, i.e the hours >= len(p.To),
+			// set the sharehours as len(p.To). If we don't have enough hours, i.e. hours < len(p.To),
+			// share all hours we have.
```
Comment (gz-c) [inline src/transaction/create.go]: This should consider if there is change, then the change output should be guaranteed at least one hour, if sharefactor < 1

**Comment by iketheadore** [comment]
Yes, this change will affect the `/api/v2/transaction` as well, all api that call the `transaction.Create` method will be affected.

**Comment by gz-c** [comment]
Merge conflict

**Comment by gz-c** [comment]
Can you update the changelog to say that it also affects `/api/v2/transaction` endpoint

**Comment by gz-c** [comment]
I will continue this PR

**Comment by gz-c** [comment]
Decided not to implement due to the complexity for marginal benefit

**Pull request closed without merging**